pkg/transforms/policy.go

```diff
@@ -37,16 +41,129 @@ func PolicyResourceBuilder(p *p.Policy) *PolicyResource {
 	if okns && okpp {
 		node.Properties["_parentPolicy"] = pnamespace + "/" + ppolicy
 	}
+
 	return &PolicyResource{node: node}
 }
 
+// For cert, config, operator policies.
+// This function returns `annotations`, `managedFields` for `source`,
+// and `severity`, `compliant`, `responseAction`, `parentPolicyNs`, `parentPolicyName`
+// If `parentPolicyNs`, `parentPolicyName` do not exist, the policy is a stand-alone policy
+func getPolicyCommonProperties(c *unstructured.Unstructured, node Node) Node {
+	node.Properties["annotations"] = c.GetAnnotations()
+	node.Properties["managedFields"] = c.GetManagedFields()
+
+	typeMeta := v1.TypeMeta{
+		Kind:       c.GetKind(),
+		APIVersion: c.GetAPIVersion(),
+	}
+
+	apiGroupVersion(typeMeta, &node) // add kind, apigroup and version
+
+	policyNsName, ok := c.GetLabels()["policy.open-cluster-management.io/policy"]
+	if ok {
+		stringSlice := strings.Split(policyNsName, ".")
+		// If len(stringSlice) != 2, failed to parse policyNsName
+		if len(stringSlice) == 2 {
+			node.Properties["parentPolicyNs"] = stringSlice[0]
+			node.Properties["parentPolicyName"] = stringSlice[1]
+		}
+	}
+
+	complianceState, found, err := unstructured.NestedString(c.Object, "status", "compliant")
+	if !found || err != nil {
+		node.Properties["compliant"] = ""
+	} else {
+		node.Properties["compliant"] = complianceState
+	}
+
+	severity, found, err := unstructured.NestedString(c.Object, "spec", "severity")
+	if !found || err != nil {
+		node.Properties["severity"] = ""
+	} else {
+		node.Properties["severity"] = severity
+	}
+
+	// responseAction is same as `remediationAction`
```

Comment:
I'm curious: Why the change in name to `responseAction`? Is it predefined or trying to be more generic?

Reply:
`responseAction` includes gatekeeper's `enforcementAction`

Reply:
Let's not use this name for the search collector and use the real name. Response action should just be what is displayed in the new UI.